File: src/src/main/java/com/mewebstudio/translatable/entity/generator/GeneratedId.java
package com.mewebstudio.translatable.entity.generator;

import org.hibernate.annotations.IdGeneratorType;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@IdGeneratorType(ULIDGenerator.class)
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.FIELD, ElementType.METHOD})
public @interface GeneratedId {
}

File: src/main/kotlin/com/mewebstudio/translatable/entity/generator/GeneratedId.kt
package com.mewebstudio.translatable.entity.generator

import org.hibernate.annotations.IdGeneratorType

@IdGeneratorType(ULIDGenerator::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.FIELD)
annotation class GeneratedId
